### PR TITLE
chore: test on python 3.13

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

1. Tests against Python 3.13
1. Install types dynamically with mypy, this is more inline with what we do internally (removes explicitly defined ones from setup.py)
1. Removes `twine` since we now use GitHub Actions to release, that package is not needed since it's already contained in the action we use

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,9 +20,6 @@
     "pytest-cov==4.*",
     "pytest-vcr==1.*",
     "pytest==7.*",
-    "twine==5.*;python_version>='3.8'",  # TODO: remove python pin when 3.7 is dropped
-    "types-requests",
-    "types-urllib3",
```
Comment (jchen293) [inline setup.py:25]: Why was this removed?

**Review by jchen293 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (Justintime50) [reply setup.py:25]: Described in the PR summary.
